vowpalwabbit/memory.h

```diff
@@ -21,4 +21,54 @@ T* calloc_or_throw(size_t nmemb)
 template<class T> T& calloc_or_throw()
 { return *calloc_or_throw<T>(1); }
 
+template<class T>
```

Comment:
You might consider wrapping the whole thing in the `#ifdef`, and alias it to `calloc_or_throw` (with `#define`) if the constant isn't defined.